include/asterisk/res_pjsip_pubsub.h

```diff
@@ -282,6 +294,25 @@ struct ast_sip_notifier {
 	 * \retval -1 Failure
 	 */
 	int (*subscription_established)(struct ast_sip_subscription *sub);
+	/*!
+	 * \brief Called when a SUBSCRIBE arrives for an already active subscription.
+	 *
+	 * \param sub The existing subscription
+	 * \retval 0 Success
+	 * \retval -1 Failure
+	 */
+	int (*refresh_subscribe)(struct ast_sip_subscription *sub, pjsip_rx_data *rdata);
+	/*!
+	 * \brief Optional callback to execute before sending outgoing NOTIFY responses.
+	 *        Because res_pjsip_pubsub creates the tdata internally, this allows modules
+	 *        to access the tdata if needed, e.g. to add custom headers.
+	 *
+	 * \param sub The existing subscription
+	 * \param tdata The pjsip_tx_data to use for the outgoing NOTIFY
+	 * \retval 0 Success
+	 * \retval -1 Failure
+	 */
+	int (*tdata_callback)(struct ast_sip_subscription *sub, pjsip_tx_data *tdata);
```

Comment:
tdata_callback is not a very descriptive name in comparison to the others, something indicating the callback is before the NOTIFY is sent would be better such as:

notify_created